blog/content/post/2020-02-19-opnsense-xen-trenchboot-demo.md

```diff
@@ -0,0 +1,171 @@
+which allows to build **High performance secure firewall with native WiFi
+performance**.
+
+Each section of this article describe each requirement, we set ourselves, in
+details. Whereas last chapter introduce our recent results with demo and
+performance analysis.
+
+## coreboot + TrenchBoot = verified and trusted boot
+
+Our first goal is to provide low-level security. One step to achieve that is to
+use open source firmware. Second step is to enable all possible mechanisms which
+allow to somehow measure and verify boot process. Third step is to join them
+together and enjoy trusted platform. Seems easy, right? Actually, entire process
+is more complicated and complex. For the purpose of this blog post I will
+shortly introduce what tools and techniques we have used to meet first
+requirement.
+
+#### coreboot configuration
+
+Our firmware is `coreboot` with enabled `vboot` feature. That configuration is a
+way of implementation of **Static Root of Trust for Measurements** (SRTM). It is
+a technology which measures and verifies device firmware components during boot
+process. That components are in fact piece of codes of specific part of system.
+Measurements of each are stored in Trusted Platform Module (TPM) special
+registers. After coreboot ends its execution, firmware is being verified and if
+it is not corrupted, next stage of booting starts.
+
+>For details about SRTM I refer to articles and papers available in Internet
+(e.g. [this
+one](http://security.hsr.ch/mse/projects/2011_Root_of_Trust_for_Measurement.pdf))
+
+#### TrenchBoot
+
+TrenchBoot is the next stage running after coreboot execution ended. Security
+concept is exactly the same as in previous case - measure and verify running
+environment. TrenchBoot uses slightly different technology called **Dynamic Root
+of Trust for Measurements** (DRTM). Pointing out differences is not a subject of
+this article. To be honest, it is very complex topic and understanding it could
+take too much time. What you need to know (or trust us about) is that other
+TPM's registers store those measurements and running environment is verified and
+can be described as trusted.
+
+Below diagram presents described technologies and features. It is simplified,
+but should give you main idea about their usage and role.
+
+![coreboot_trenchboot](../../static/img/coreboot_trenchboot.png)
+
+## Virtualization, isolation and WiFi performance
+
+If virtualization, hypervisor, Xen, Dom0 and related terminology is foreign to
+you, please get familiar with those blog posts first:
+
+- [Xen Project](https://blog.3mdeb.com/2020/2020-02-05-meta-pcengines-xen/)
+- [pfSense under Xen - part 1](https://blog.3mdeb.com/2019/2019-11-06-pfsense-under-xen-introduction/)
+- [pfSense under Xen - part 2](https://blog.3mdeb.com/2019/2019-12-13-pfsense-boot-under-xen/)
+
+When you already have basic knowledge, we can move on to the essential part of
+this section.
```

Comment:
@piotr-kleins I believe here we should have a summary of concepts explained in other posts. The methodology that you use will not work since people typically do not have enough time to dig in into all that stuff. Please present summary and mention that if you need deep dive you can find it in above posts.